### DISCUSSION TITLE
Using RNVSimpleCache.INSTANCE for HlsMediaSource.Factory (HLS Caching on Android)

### DISCUSSION THREAD

**Discussion by IslamRustamov** [TheWidlarzGroup/react-native-video#4271] [Q&A] [answered]
### What are you trying to do?

Enable HLS video caching on Android. As we know - this library does not support HLS video caching on Android.

So, I dug deeper into the source code of this lib and found out in **ReactExoplayerView.java** in **buildMediaSource** method (line 1053) this line can be changed to use `RNVSimpleCache.INSTANCE.getCacheFactory`.

After this change - i checked network inspector in Android Studio and saw that my HLS videos are not being downloaded again (presumably, they are being saved to cache).

So, why didn't authors of this package made this available for HLS video caching if it seems rather easy to do? What are the drawbacks of this? (I'm not proficient in Android development, so maybe the change I did is a bad thing to do)

### What have you tried?

Changed the `HlsMediaSource.Factory` construction in `buildMediaSource` (ReactExoplayerView.java, line 1053) from the default `mediaDataSourceFactory` to `RNVSimpleCache.INSTANCE.getCacheFactory(buildHttpDataSourceFactory(true))`, then verified via the Android Studio network inspector that the HLS videos are not being re-downloaded.

### Relevant code

Original line:

```
mediaSourceFactory = new HlsMediaSource.Factory(
    mediaDataSourceFactory
).setAllowChunklessPreparation(source.getTextTracksAllowChunklessPreparation());
```

Changed to:

```
mediaSourceFactory = new HlsMediaSource.Factory(
        RNVSimpleCache.INSTANCE.getCacheFactory(buildHttpDataSourceFactory(true))
).setAllowChunklessPreparation(source.getTextTracksAllowChunklessPreparation());
```

### Platforms

Android

**Comment by freeboub** [comment, marked as answer]
If I remember well the SimpleCache was working only with mp4. I reviewed the doc again and I don't see this limitation anymore! So I think your change can be integrated without problem! If you want to create a pull request with this change, it's welcome!

**Reply by IslamRustamov** [reply]
Thanks! Check it out https://github.com/TheWidlarzGroup/react-native-video/pull/4272
